sceptre/cli/helpers.py

```diff
@@ -173,6 +189,23 @@ def _generate_text(stream):
     return stream
 
 
+def process_template(
+    template: dict, output_format: str, type: str = "template"
+) -> None:
+    """
+    Helper to write templates and configs used in dump commands.
+
+    :param template: the template.
```

Comment:
You'll have to elaborate ....

Reply:
Your `template` variable here isn't actually a template. In other words, your variable naming lies. For a helper function that's meant to stand on its own, the interface must be clear, simple, and functional. But you're making a lot of assumptions here about what `template` is. 

It should take a `stack_name` and a `template_dict` parameter instead of a dict that happens to have a key that is the stack name and a value that is the actual template dict. Your code would be simpler, more readable, and easier to interact with if you did that.

Reply:
I'm not convinced that a variable named `template_dict` of type `dict` is clearer than `template: dict` and to my mind the whole comes out more rather than less messy afterwards. Perhaps we are better off without this helper at all and I revert to how it was before?

Reply:
For example on the caller side:

```python
    for template in output:
        dump_to_file(list(template.keys())[0], template, output_format)
```

We don't want duplication of `list(template.keys())[0]` all over the place for the sake helping the reader of that helper function since most people are going to be reading the CLI functions; not the helpers.

Reply:
My point is that you're calling the dict a "template", but it's actually a dict with a single key indicating the stack name and a its value is the actual template. In other words, your variable name is wrong and the interface of the function is more challenging than it ought to be. 

Rather than assuming this arbitrary structure that is a mouthful to describe, I'm just saying take two parameters instead of this complex dict structure that is isn't well-typed.